airbyte-json-validation/src/main/java/io/airbyte/validation/json/ConfigJsonSchemaValidator.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.validation.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.Set;
+
+public interface ConfigJsonSchemaValidator<T extends Enum<T>> {
```

Comment:
This indirection feels artificial. It is basically duplicating the `JsonSchemaValidator.java` interface

Reply:
I would rename it to `ConfigSchemaValidator`

Reply:
It is definitely artificial. Empirically we are not validating these configurations where we should. My hypothesis is because it's a hassle to do so and costs a lot of code bloat, so i'm just trying to make it easier to do the right thing. If you have a thought on how to make it less artificial, lmk and i can adjust. 

I picked this abstraction because we have a convention for defining json schemas and connecting them to an enum. This abstraction is made to make it very easy to validate easily for anything following that convention.